frontend/src/utils/DateUtils.ts

```diff
@@ -8,3 +8,25 @@ export const formatDateString = (dateStringInput: Date): string => {
     timeZone: "UTC",
   });
 };
+
+export const formatDateStringWithYear = (dateStringInput: string): string => {
+  const date = new Date(dateStringInput);
+  return date.toLocaleDateString("en-US", {
+    weekday: "long",
+    month: "short",
+    day: "2-digit",
+    year: "numeric",
+    timeZone: "UTC",
+  });
+};
+
+// currently only supports filter by week and month
+export const dateInRange = (start: string, filterType: string): boolean => {
```

Comment:
```suggestion
export const dateInRange = (start: string, filterType: FilterType): boolean => {
```

Let's move the `FilterType` definition into a new file under `frontend/src/types/DateFilterTypes.ts` so it can be shared by this util and the postings page component